storage/volume/core.go

```diff
@@ -272,29 +274,40 @@ func (c *Core) ListVolumes(labels map[string]string) ([]*types.Volume, error) {
 		retVolumes = append(retVolumes, v)
 	}
 
-	// filter volumes if specify labels
-	if len(labels) != 0 {
-		filteredVolumes := make([]*types.Volume, 0)
-		for k, v := range labels {
-			for _, vol := range retVolumes {
-				if val, ok := vol.Labels[k]; ok && val == v {
-					filteredVolumes = append(filteredVolumes, vol)
-				}
-			}
-		}
+	// filter volumes
+	if filter.Len() == 0 {
+		return retVolumes, nil
+	}
 
-		return filteredVolumes, nil
+	filteredVolumes := make([]*types.Volume, 0)
+	for _, vol := range retVolumes {
+		// do label filter
```

Comment:
just for confirm, filter only support `and` right?

Reply:
relationship between different filter was `and`, but if multi same filter was use, such as `--driver=A, --driver=B`, there will be A or B.

Reply:
good to know. Thanks